src/rendering/renderers/shared/shader/UboSystem.ts

```diff
@@ -21,7 +21,7 @@ export class UboSystem implements System
     /** Cache of uniform buffer layouts and sync functions, so we don't have to re-create them */
     private _syncFunctionHash: Record<string, {
         layout: UboLayout,
-        syncFunction: (uniforms: Record<string, any>, data: Float32Array, offset: number) => void
+        syncFunction: (uniforms: Record<string, any>, data: Float32Array, dataInt32: Int32Array, offset: number) => void
```

Comment:
Then we'll have to rename it in many other places. Also, Buffer `data` sometimes isnt Float32Array because people make different stuff there. Its just it is Float32Array in this particular place

Reply:
should we pass in the buffer here, rather than the two views? This depends on if there are other view we might want to add in the future lke f64 / u32?